.github/workflows/ci.yml

```diff
@@ -0,0 +1,151 @@
+        if: startsWith(matrix.os, 'ubuntu')
+        run: docker run --detach --name postgresql -p 5432:5432 npgsql/postgres:${{ matrix.pg_major }}
+        shell: bash
+
+      - name: Start PostgreSQL ${{ matrix.pg_major }} (Windows)
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          # Install PostgreSQL
+          curl -o pgsql.zip -L https://get.enterprisedb.com/postgresql/postgresql-${{ matrix.edb_version }}-windows-x64-binaries.zip
+          unzip pgsql.zip -x 'pgsql/include/**' 'pgsql/doc/**' 'pgsql/pgAdmin 4/**' 'pgsql/StackBuilder/**'
+
+          # Match Npgsql CI Docker image and stash one level up
+          cp {$GITHUB_WORKSPACE/.build/docker,pgsql}/server.crt
+          cp {$GITHUB_WORKSPACE/.build/docker,pgsql}/server.key
+
+          # Install PostGIS
+          POSTGIS_FILE="postgis-bundle-pg${{ matrix.pg_major }}-${{ matrix.postgis_version }}x64"
+          curl -o postgis.zip -L https://download.osgeo.org/postgis/windows/pg${{ matrix.pg_major }}/${POSTGIS_FILE}.zip
+          unzip postgis.zip -d postgis
+          cp -a postgis/$POSTGIS_FILE/. pgsql/
+
+          # Start PostgreSQL
+          pgsql/bin/initdb -D pgsql/PGDATA -E UTF8 -U postgres
+          pgsql/bin/pg_ctl -D pgsql/PGDATA -l logfile -o '-c max_prepared_transactions=10 -c ssl=true -c ssl_cert_file=../server.crt -c ssl_key_file=../server.key' start
+
+          # Configure test account
+          pgsql/bin/psql -U postgres -c "CREATE ROLE npgsql_tests SUPERUSER LOGIN PASSWORD 'npgsql_tests'"
+          pgsql/bin/psql -U postgres -c "CREATE DATABASE npgsql_tests OWNER npgsql_tests"
+        shell: bash
+
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1.0.2 # Issues with DOTNET_ROOT
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+
+      - name: Test (EFCore.PG.Tests)
+        run: dotnet test test/EFCore.PG.Tests --configuration Debug
+        shell: bash
+
+      - name: Test (EFCore.PG.FunctionalTests)
+        run: dotnet test test/EFCore.PG.FunctionalTests --configuration Debug
+        shell: bash
+
+      - name: Test (EFCore.PG.Plugins.FunctionalTests)
+        run: dotnet test test/EFCore.PG.Plugins.FunctionalTests --configuration Debug
+        shell: bash
+
+  publish:
+    needs: build
+
+    runs-on: windows-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Pack NuGet packages (CI versions)
+        run: |
+          VERSION_SUFFIX="ci.$(date -u +%Y%m%dT%H%M%S)+sha.${GITHUB_SHA:0:9}"
+          dotnet pack src/EFCore.PG          --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX
+          dotnet pack src/EFCore.PG.NodaTime --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX
+          dotnet pack src/EFCore.PG.NTS      --configuration Release --output nupkgs/ci --version-suffix $VERSION_SUFFIX
+        shell: bash
+
+      - name: Pack NuGet packages (Release versions)
```

Comment:
Ideally we'd do this only for released-tagged versions... In appveyor we actually parse the tag, check if it's vX.Y.Z, and if so, enter "release mode". It's not critical (definitely not in a V1), but it we do this we can do automatic creation of the github release and maybe even automatic upload to nuget.org.

Reply:
At some point I had a thought that having release version artifacts for each CI build could be interesting (e.g. pretend like these are the real `v3.1.0`).

But in retrospect I'm neither sure that's a good idea, nor that it's worth the extra CI step. I'll look into adding some logic to the previous step so that `VERSION_SUFFIX` is set null when necessary

Reply:
`--version-suffix` requires an argument, so kept separate steps but made them conditional on the branch. Need to take another look at this in the context of real releases.